backend/ml/inputs.py

```diff
@@ -41,63 +46,75 @@ def ratings_properties(self) -> pd.DataFrame:
-        ].factorize()
+    def __init__(self, dataset_zip: Union[str, BinaryIO]):
+        if isinstance(dataset_zip, str) and (
+            dataset_zip.startswith("http://") or dataset_zip.startswith("https://")
+        ):
+            dataset_zip, _headers = urlretrieve(dataset_zip)  # nosec B310
+
+        with zipfile.ZipFile(dataset_zip) as zip_file:
+            zip_root_dir = next(path for path in zipfile.Path(zip_file).iterdir() if path.is_dir())
+
+            with (zip_root_dir / "comparisons.csv").open() as comparison_file:
+                self.comparisons = pd.read_csv(comparison_file)
+                self.comparisons.rename(
+                    {"video_a": "entity_a", "video_b": "entity_b"}, axis=1, inplace=True
+                )
 
-    def get_comparisons(
-        self, trusted_only=False, criteria=None, user_id=None
-    ) -> pd.DataFrame:
-        dtf = self.public_dataset.copy(deep=False)
+            with (zip_root_dir / "users.csv").open() as users_file:
+                self.users = pd.read_csv(users_file)
+                self.users.index.name = "user_id"
+
+            username_to_user_id = pd.Series(
+                data=self.users.index, index=self.users["public_username"]
+            )
+            self.comparisons = self.comparisons.join(username_to_user_id, on="public_username")
+
+    def get_comparisons(self, criteria=None, user_id=None) -> pd.DataFrame:
+        dtf = self.comparisons.copy(deep=False)
         if criteria is not None:
             dtf = dtf[dtf.criteria == criteria]
         if user_id is not None:
             dtf = dtf[dtf.user_id == user_id]
+        dtf["weight"] = 1
         return dtf[["user_id", "entity_a", "entity_b", "criteria", "score", "weight"]]
 
     @cached_property
     def ratings_properties(self):
-        # TODO support trust_scores from the public dataset
         user_entities_pairs = pd.Series(
             iter(
-                set(self.public_dataset.groupby(["user_id", "entity_a"]).indices.keys())
-                | set(
-                    self.public_dataset.groupby(["user_id", "entity_b"]).indices.keys()
-                )
+                set(self.comparisons.groupby(["user_id", "entity_a"]).indices.keys())
+                | set(self.comparisons.groupby(["user_id", "entity_b"]).indices.keys())
             )
         )
         dtf = pd.DataFrame([*user_entities_pairs], columns=["user_id", "entity_id"])
         dtf["is_public"] = True
-        top_users = dtf.value_counts("user_id").index[:6]
-        dtf["is_trusted"] = dtf["is_supertrusted"] = dtf["user_id"].isin(top_users)
+        dtf["trust_score"] = dtf["user_id"].map(self.users["trust_score"])
+        scaling_calibration_user_ids = (
+            dtf[dtf.trust_score > self.SCALING_CALIBRATION_MIN_TRUST_SCORE]["user_id"]
+            .value_counts(sort=True)[: self.MAX_SCALING_CALIBRATION_USERS]
+            .loc[lambda count: count >= self.SCALING_CALIBRATION_MIN_ENTITIES_TO_COMPARE]
```

Comment:
I'm not sure this ".loc" does what we want.

For videos, the constraint SCALING_CALIBRATION_MIN_ENTITIES_TO_COMPARE isn't effective. Because there is at least one contributor that compared more than SCALING_CALIBRATION_MIN_ENTITIES_TO_COMPARE entities (cf. get_scaling_calibration_users).

And for the elections poll, we don't have SCALING_CALIBRATION_MIN_ENTITIES_TO_COMPARE entities, so this would return 0 scaling contributors.

Reply:
I wanted to replicate the definition used in production, but I agree that does not really make sense on the public dataset. I will remove the filter on the number of compared entities.